cel/canonical_eventlog.go

```diff
@@ -121,9 +131,35 @@ type Content interface {
 	GetTLV() (TLV, error)
 }
 
-// CEL represents a Canonical Eventlog, which contains a list of Records.
-type CEL struct {
-	Records []Record
+// CEL represents a Canonical Event Log, which contains a list of Records.
+type CEL interface {
+	// Records returns all the records in the CEL.
+	Records() []Record
+	// AppendEvent appends a new record to the CEL.
+	AppendEvent(Content, []crypto.Hash, int, MRExtender) error
+	// EncodeCEL returns the TLV encoding of the CEL.
+	EncodeCEL(*bytes.Buffer) error
+	// Replay verifies the contents of the event log with the given MR bank.
+	Replay(register.MRBank) error
+	// MRType returns the measurement register type used in the CEL.
+	MRType() MRType
+}
+
+// eventLog represents a Canonical Event Log, which contains a list of Records.
+type eventLog struct {
+	Recs []Record
+	Type MRType
```

Comment:
I think technically one eventlog can support more than one MRType. Some records measured to PCRs, some records measured to RTMR, while both events measured to the same eventlog.

But for our current use case, I think this is fine to restrict one MRType for one eventlog.